source/MicroBitAccelerometer.cpp

```diff
@@ -71,34 +68,21 @@ Accelerometer& MicroBitAccelerometer::autoDetect(MicroBitI2C &i2c)
     
     if (!autoDetectCompleted)
     {
-        MicroBitAccelerometer::detectedAccelerometer = NULL;
-        MicroBitCompass::detectedCompass = NULL;
+        MicroBitAccelerometer::driver = NULL;
+        MicroBitCompass::driver       = NULL;
 
-        // All known accelerometer/magnetometer peripherals have the same alignment
-        if (FXOS8700::isDetected(i2c, 0x3E))
-        {
-            FXOS8700 *fxos = new FXOS8700(i2c, irq1, coordinateSpaceFXOS8700, 0x3E);
-            MicroBitAccelerometer::detectedAccelerometer = fxos;
-            MicroBitCompass::detectedCompass = fxos;
-        }
+        // Now, probe for the LSM303, and if it doesn't reply, panic
+        if (!LSM303Accelerometer::isDetected(i2c, 0x32))
+            microbit_panic( 50 ); // FixMe: This should be a constant from the PanicCode enum, but this is out of sync
```

Comment:
@JohnVidler I think DAL delays the panic until the accelerometer is used (https://github.com/lancaster-university/microbit-dal/commit/96126fd23f3147257c62cd1aa98a26588c869d63), so that an unresponsive accelerometer might not stop all programs running. I'm not sure if this idea was implemented in CODAL. It looks like CODAL simply didn't panic.

Reply:
Good catch, this is now addressed in subsequent patches.